vm/state_manager/src/errors.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use amt::Error as AmtErr;
+use std::fmt;
+
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    /// Error orginating from state
+    State(String),
+    /// Error originating from amt
+    AMT(String),
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Error::State(msg) => write!(f, "Error state data: {}", msg),
+            Error::AMT(msg) => write!(f, "Error originating from the AMT: {}", msg),
```

Comment:
Here is another example of appending strings to error messages, maybe consider not adding the prefix to the error message for these to remove redundant messages?